stdlib/public/core/FloatingPoint.swift

```diff
@@ -328,15 +328,20 @@ public protocol FloatingPoint: SignedNumeric, Strideable, Hashable
   /// `infinity` is greater than this value.
   static var greatestFiniteMagnitude: Self { get }
 
-  /// The mathematical constant pi.
+  /// The [mathematical constant π][wiki], approximately equal to 3.14159.
   ///
-  /// This value should be rounded toward zero to keep user computations with
-  /// angles from inadvertently ending up in the wrong quadrant. A type that
-  /// conforms to the `FloatingPoint` protocol provides the value for `pi` at
-  /// its best possible precision.
+  /// A common application of this constant in geometry is to measure a half
```

Comment:
Close! I think we can refine this sentence a bit perhaps:

It's unclear why it should matter to the reader that measuring a half revolution in radians is "a common application." That it's an important fact is evident from being mentioned in documentation of all the other applications, but whether it's common has no bearing on the information being conveyed. The sentence also implies that we are talking about angles, or revolutions _of a circle_, but omits the word "circle," instead hinting at it by specifying that we're talking about an application of pi "in geometry"—which would be unnecessary to mention if we just talked about angles or a circle.

The sentence also _implies_, rather than states outright, the central point of the paragraph, which is that half a revolution is _pi_ radians. Instead, it states that "this constant" is an "application" of measuring half a revolution, then in the following sentence that "it" (i.e., half a revolution; although grammatically "it" would have to be taken as the subject of the previous sentence—namely, the "common application") is equivalent to 180°, but the relation that half a revolution is equivalent to 180° isn't the key point here.

Therefore, I'd suggest something like, "When measuring angles, a complete revolution (360°) is 2π radians."

Reply:
> It's unclear why it should matter to the reader that measuring a half revolution in radians is "a common application."

I think I know what you mean, but I'll argue that it primes the reader for remembering this constant when dealing with geometry code. Rather than just presenting factual information, it helps the reader develop a more efficient internal model for working with this library. If someone finds this paragraph by inspecting `pi` from pre-existing code, it's fine to just be descriptive. But if someone is browsing the documentation to get a sense of it so they can write new code from scratch, it pays to be prescriptive and nudge developers towards making the best use of the standard library.

> The sentence also implies that we are talking about angles, or revolutions of a circle, but omits the word "angle" or "circle,"

Fair, in my defense this role was previously satisfied by the paragraph about the definition of pi. Will try to make it more explicit, and also focus more on the key point.

Reply:
See if it is more to the point now.

Reply:
I like it. Last nit is that “this constant” could be interpreted as either π or `.pi`—one is an approximation of the other, and depending on the precision and the operation in question the latter may or may not be equivalent to using 180 degrees. Therefore, I’d specifically write “π” rather than “this constant.”

Reply:
That's actually a good point, thanks for flagging.